core/src/test/scala/io/snappydata/app/CreateIndexTest.scala

```diff
@@ -138,12 +586,62 @@ class CreateIndexTest extends SnappyFunSuite {
     doPrint("Drop Index - Done")
   }
 
-  def verifyRows(r: Row) : Unit = {
+  def verifyRows(r: Row): Unit = {
     doPrint(r.toString())
     assert(r.toString() == "[111]", "got=" + r.toString() + " but expected 111")
   }
 
   def doPrint(s: Any): Unit = {
-    //println(s)
+    // println(s)
+  }
+
+  private case class QueryExecutor(snContext: SnappyContext,
+      showResults: Boolean = false,
+      withExplain: Boolean = false) {
+
+    private[app] def apply(sqlText: String, explainQ: Boolean = false)
+        (implicit validate: DataFrame => Unit = _ => ()) = {
+      val msg = s"Executing $sqlText"
+      logInfo(msg) // log it
+      info(msg) // inform it
+
+      val selectRes = snContext.sql(sqlText)
+
+      if (withExplain || explainQ) {
+        selectRes.explain(true)
+      }
+
+      validate(selectRes)
+
+      if (showResults) {
+        selectRes.show
+      }
+
+      selectRes
+    }
+
   }
+
+  private def validateIndex(index: Seq[String], tables: String*)(df: DataFrame): Unit = {
+    val (indexesMatched, indexesUnMatched) = df.queryExecution.optimizedPlan.collect {
```

Comment:
I suppose optimizedPlan will only have IndexColumnRel which gets translated into ColumnTableScan within SparkPlan and is hard to distinguish b/w table vs index scan. Yes I agree we are really avoiding exchange check should be there. One of the item in my todo list :)